### PR TITLE
Implement EIP-4844 methods in `RespondingEth2Peer` (used in tests)

### PR DESCRIPTION
**Pull request by StefanBratanov**
## PR Description
Will help for EIP-4844 tests using the `RespondingEth2Peer`

## Fixed Issue(s)
N/A

## Documentation

- [X] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [X] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: networking/eth2/src/testFixtures/java/tech/pegasys/teku/networking/eth2/peers/RespondingEth2Peer.java [hunk 1]
```diff
@@ -199,78 +199,117 @@ public SafeFuture<Void> requestBlocksByRange(
...
+
+    return createPendingBlockAndBlobsSidecarRequest(handler);
   }
 
-  private <T> SafeFuture<T> createPendingRequest(PendingBlockRequestHandler<T> handler) {
-    final PendingBlockRequestHandler<T> filteredHandler =
-        PendingBlockRequestHandler.filterRequest(handler, blockRequestFilter);
-    final PendingBlockRequest<T> request = new PendingBlockRequest<>(filteredHandler);
+  private <T> SafeFuture<T> createPendingBlockRequest(
+      PendingRequestHandler<T, SignedBeaconBlock> handler) {
```
Comment (zilm13) [inline networking/eth2/src/testFixtures/java/tech/pegasys/teku/networking/eth2/peers/RespondingEth2Peer.java]: final
+next 2 methods

File: networking/eth2/src/testFixtures/java/tech/pegasys/teku/networking/eth2/peers/RespondingEth2Peer.java [hunk 2]
```diff
@@ -436,67 +485,97 @@ public void complete() {
     }
   }
 
-  private interface PendingBlockRequestHandler<T> {
+  private interface PendingRequestHandler<T, S> {
```
Comment (zilm13) [inline networking/eth2/src/testFixtures/java/tech/pegasys/teku/networking/eth2/peers/RespondingEth2Peer.java]: Takes a bit to understand what is it for, especially as S is now for a type and T for response future if I'm getting it right. I'd use long names like in `sendRequest` or at least something like RT, T (though looks bad too)

**Review by zilm13 — APPROVED** [review]
LGTM, just a few things

File: networking/eth2/src/testFixtures/java/tech/pegasys/teku/networking/eth2/peers/RespondingEth2Peer.java (hunk 2 above)
Comment (StefanBratanov) [reply networking/eth2/src/testFixtures/java/tech/pegasys/teku/networking/eth2/peers/RespondingEth2Peer.java]: Makes sense, changed it to `ResponseT` and `HandlerT`

**Pull request merged**